src/p2p/net_node.cpp

```diff
@@ -37,9 +37,13 @@ namespace nodetool
     const command_line::arg_descriptor<std::string, false, true> arg_p2p_bind_port = {
         "p2p-bind-port"
       , "Port for p2p network protocol"
-      , cryptonote::arg_testnet_on
-      , std::to_string(config::testnet::P2P_DEFAULT_PORT)
       , std::to_string(config::P2P_DEFAULT_PORT)
+      , cryptonote::arg_testnet_on
+      , [](bool testnet, bool defaulted, std::string val) {
+          if (testnet && defaulted)
+            return std::to_string(config::testnet::P2P_DEFAULT_PORT);
+          return val;
+        }
```

Comment:
The asymmetry of this feels iffy somehow, though I can see how it works. I don't feel bad enough about it to ask for a change though.

Reply:
Yes, it is indeed much less clean than the previous version, but lambdas seem to be the most transparent way to statically configure complete runtime dependencies (that is, not only for default values).

Reply:
I don't mind lambas, just the asymmetry of the code feels odd.